Tools/Scripts/layout-test-failure-analysis

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import logging
+import os
+import sys
+
+import re
+import requests
+from urllib.parse import urlparse
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Export test names in the order they were run for a specific worker from stdio input.')
+
+    # data source
+    input_group = parser.add_mutually_exclusive_group(required=True)
+    input_group.add_argument('-f', '--file',
+                        dest='input_file',
+                        help='Specifies the file to read for layout test results data.',
+                        type=str)
+    input_group.add_argument('-u', '-url',
+                        dest='input_url',
+                        help='Specifies the URL from which to retrieve layout test results data.',
+                        type=str)
```

Comment:
Do we really need two different arguments here? Can't well tell the difference based on the URL having :// in it?

Generally, it's a bit of a "code smell" when we have a mandatory script argument passed via '-<something>'. That's not a hard-and-fast rule, but in this case, it does seem like a single line of Python code could differentiate between a URL and file.

Reply:
Thanks @briannafan for showing me the wonderful world of positional arguments!